### PR TITLE
Add option to create drydep velocity plots from benchmark output

### LINKED ISSUE #273: [FEATURE REQUEST] Add dry deposition velocities to benchmark output
Mat Evans has requested we include annual mean dry deposition velocity output for key species (e.g. O3, NH4, HNO3, Acetone, SO4, NIT, NH4) in the 1-month and 1-year benchmark output to track changes to the dry deposition. These output are already saved out in the benchmark simulations. We will need to update gcpy to either plot or tabulate the data and compare to the previous version.

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://gcpy.readthedocs.io/en/stable/reference/CONTRIBUTING.html)

### Describe the update

This is the companion PR to #273.  We have added the option to plot drydep velocities as part of the standard benchmark outputs.

Updates in this version:
- Added script `gcpy/benchmark/modules/benchmark_utils.py` with a few utility functions.  
  - This is where we can continue to abstract utility function specific to benchmark plotting codes from `gcpy/util.py` and `gcpy/benchmark_funcs.py`.  
  - Eventually, all of the benchmarking code will be moved from `gcpy/gcpy` to `gcpy/benchmark/*`, so as to isolate them from other user functionality.

- Added script `gcpy/benchmark/modules/benchmark.drydep.py`, which contains function `make_benchmark_drydep_plots`.

- Modified the following scripts to call `make_benchmark_drydep_plots` from `benchmark_drydep.py`:
  - `gcpy/benchmark/run_benchmark.py`
  - `gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py`

- Added the `plot_drydep:True` YAML tag to:
  - `gcpy/benchmark/config/1yr_fullchem_benchmark.yml` 

- Added the `plot_drydep: False` YAML tag to:
  - `gcpy/benchmark/cloud/template.1hr_benchmark.yml`
  - `gcpy/benchmark/cloud/template.1mo_benchmark.yml`
  - `gcpy/benchmark/config/1yr_ftt_benchmark.yml`
  - `gcpy/benchmark/config/1mo_fullchem_benchmark.yml`

- Changed `operations_budget: False` to `ops_budget_table: False` in `gcpy/benchmark/config/1yr_tt_benchmark.yml`.  
  - This is to match the nomenclature used in the fullchem YAML files.

### Expected changes
- By default, drydep velocity plots are ENABED from in 1-yr fullchem benchmark output.
- By default, drydep velocity plots are DISABLED in 1-month and 1-hour fullchem benchmark output. 
  - Enabling this would require an update in the GEOS-Chem repository to archive the DryDep collection when running 1-month benchmark simulations.

### Related Github Issue(s)

- Closes #273, based on a suggestion by @mje1398

### CODE REVIEW

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py [hunk 1]
```diff
@@ -1194,6 +1226,66 @@ def gcc_vs_gcc_ops_budg(mon):
                     n_job=config["options"]["n_cores"]
                 )
 
+        # ==================================================================
+        # GCC vs. GCC drydep plots
```
Comment (msulprizio) [inline gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:1230]: This should be "GCHP vs GCC"

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py [hunk 2]
```diff
@@ -1194,6 +1226,66 @@ def gcc_vs_gcc_ops_budg(mon):
...
+        # ==================================================================
+        if config["options"]["outputs"]["plot_drydep"]:
+            print("\n%%% Creating GCHP vs. GCC drydep plots %%%")
+
+            # --------------------------------------------------------------
+            # GCC vs GCC drydep plots: Annual mean
```
Comment (msulprizio) [inline gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:1236]: This should be "GCHP vs GCC"

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py [hunk 3]
```diff
@@ -1194,6 +1226,66 @@ def gcc_vs_gcc_ops_budg(mon):
...
+            # --------------------------------------------------------------
+
+            # Filepaths
+            ref = get_filepaths(
+                gcc_vs_gcc_refdir,
```
Comment (msulprizio) [inline gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:1241]: Should this be `gchp_vc_gcc_refdir`?

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py [hunk 4]
```diff
@@ -1194,6 +1226,66 @@ def gcc_vs_gcc_ops_budg(mon):
...
+                subdst="AnnualMean",
+                time_mean=True,
+                weightsdir=config["paths"]["weights_dir"],
+                overwrite=True,
+                spcdb_dir=spcdb_dir,
+                n_job=config["options"]["n_cores"]
+            )
+
+            # --------------------------------------------------------------
+            # GCC vs GCC drydep plots: Seasonal
```
Comment (msulprizio) [inline gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:1269]: This should be "GCHP vs GCC"

File: gcpy/benchmark/run_benchmark.py [hunk 5]
```diff
@@ -907,6 +891,35 @@ def run_benchmark_default(config):
...
             )
 
+        # ==================================================================
+        # GCHP vs GCC drydep plots
+        # ==================================================================
+        if config["options"]["outputs"]["plot_drydep"]:
+            print("\n%%% Creating GCHP vs. GCC drydep plots %%%")
+
+            # Filepaths
+            ref = get_filepath(gcc_vs_gcc_refdir, "DryDep", gcc_ref_date)
```
Comment (msulprizio) [inline gcpy/benchmark/run_benchmark.py:901]: Should this be `gchp_vs_gcc_refdir`?

**Review by msulprizio — CHANGES_REQUESTED** [review]
I think there are some typos in the GCHP vs GCC plots for the drydep option that need to be resolved before merging.

File: gcpy/benchmark/run_benchmark.py (hunk 5 above)
Comment (yantosca) [reply gcpy/benchmark/run_benchmark.py]: Now fixed in commit 26b597e

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py (hunk 4 above)
Comment (yantosca) [reply gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py]: Now fixed in commit 26b597e

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py (hunk 3 above)
Comment (yantosca) [reply gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py]: Now fixed in commit 26b597e

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py (hunk 2 above)
Comment (yantosca) [reply gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py]: Now fixed in commit 26b597e

**Review by msulprizio — APPROVED** [review]
Looks good to merge now!

**Pull request merged**